sgv2-graphqlapi/src/main/java/io/stargate/sgv2/graphql/impl/GraphqlServiceServer.java

```diff
@@ -0,0 +1,169 @@
+import io.stargate.sgv2.common.http.CreateStargateBridgeClientFilter;
+import io.stargate.sgv2.common.http.StargateBridgeClientJerseyFactory;
+import io.stargate.sgv2.graphql.resources.HealthResource;
+import io.stargate.sgv2.graphql.resources.MetricsResource;
+import io.stargate.sgv2.graphql.web.resources.DdlResource;
+import io.stargate.sgv2.graphql.web.resources.DmlResource;
+import io.stargate.sgv2.graphql.web.resources.GraphqlCache;
+import io.stargate.sgv2.graphql.web.resources.PlaygroundResource;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Optional;
+import javax.servlet.DispatcherType;
+import javax.servlet.FilterRegistration;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.eclipse.jetty.servlets.CrossOriginFilter;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+
+public class GraphqlServiceServer extends Application<GraphqlServiceServerConfiguration> {
+
+  public static final String MODULE_NAME = "sgv2-graphqlapi";
+
+  private final Metrics metrics;
+  private final MetricsScraper metricsScraper;
+  private final HttpMetricsTagProvider httpMetricsTagProvider;
+  private final boolean disablePlayground;
+  private final boolean disableDefaultKeyspace;
+
+  public GraphqlServiceServer(
+      Metrics metrics,
+      MetricsScraper metricsScraper,
+      HttpMetricsTagProvider httpMetricsTagProvider,
+      boolean enableGraphqlPlayground,
+      boolean disableDefaultKeyspace) {
+    this.metrics = metrics;
+    this.metricsScraper = metricsScraper;
+    this.httpMetricsTagProvider = httpMetricsTagProvider;
+    this.disablePlayground = enableGraphqlPlayground;
+    this.disableDefaultKeyspace = disableDefaultKeyspace;
+  }
+
+  @Override
+  public void run(GraphqlServiceServerConfiguration config, Environment environment) {
+
+    JerseyEnvironment jersey = environment.jersey();
+
+    StargateBridgeClientFactory clientFactory =
+        buildClientFactory(config.stargate.bridge.buildChannel(), environment);
+    jersey.register(buildClientFilter(clientFactory));
+
+    GraphqlCache graphqlCache =
+        new GraphqlCache(
+            clientFactory.newClient(BridgeConfig.ADMIN_TOKEN, Optional.empty()),
+            clientFactory.getSchema(),
+            disableDefaultKeyspace);
+
+    jersey.register(
+        new AbstractBinder() {
+          @Override
+          protected void configure() {
+            bind(metricsScraper).to(MetricsScraper.class);
+            bindFactory(StargateBridgeClientJerseyFactory.class)
+                .to(StargateBridgeClient.class)
+                // Note: this should really be `RequestScoped.class`, but using that causes HK2
+                // errors.
+                // In practice that won't make a difference because our resources look up the client
+                // at most once.
+                .in(PerLookup.class);
```

Comment:
does this mean clients are created on each request? or per resource initialization? I like more the second :blush:

Reply:
Each time the client is injected in a resource. Which thankfully happens to be once per HTTP request.

I'm not happy with this workaround, but I didn't want to spend too much time on it since we're also considering switching the web framework.